src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs

```diff
@@ -161,40 +166,84 @@ internal async Task<ApplicationParameters> CreateNewApp(TokenCredential tokenCre
         {
-            var graphServiceClient = GetGraphServiceClient(tokenCredential);
+            bool updateStatus = false;
+            if (reconcialedApplicationParameters != null)
+            {
+                var graphServiceClient = GetGraphServiceClient(tokenCredential);
 
-            var existingApplication = (await graphServiceClient.Applications
-               .Request()
-               .Filter($"appId eq '{reconcialedApplicationParameters.ClientId}'")
-               .GetAsync()).First();
+                var existingApplication = (await graphServiceClient.Applications
+                   .Request()
+                   .Filter($"appId eq '{reconcialedApplicationParameters.ClientId}'")
+                   .GetAsync()).First();
 
-            // Updates the redirect URIs
-            var updatedApp = new Application
-            {
-                Web = existingApplication.Web
-            };
-            updatedApp.Web.RedirectUris = reconcialedApplicationParameters.WebRedirectUris;
+                // Updates the redirect URIs
+                if (existingApplication.Web == null)
+                {
+                    existingApplication.Web = new WebApplication();
+                }
 
-            // TODO: update other fields. 
-            // See https://github.com/jmprieur/app-provisonning-tool/issues/10
-            await graphServiceClient.Applications[existingApplication.Id]
-                .Request()
-                .UpdateAsync(updatedApp).ConfigureAwait(false);
+                var updatedApp = new Application
+                {
+                    Web = existingApplication.Web
+                };
+
+                //update redirect uris
+                List<string> existingRedirectUris = updatedApp.Web.RedirectUris.ToList();
+                List<string> urisToEnsure = ValidateUris(toolOptions.RedirectUris).ToList();
+                existingRedirectUris.AddRange(urisToEnsure);
+                updatedApp.Web.RedirectUris = existingRedirectUris.Distinct();
+
+                //update implicit grant settings
+                if (updatedApp.Web.ImplicitGrantSettings == null)
+                {
+                    updatedApp.Web.ImplicitGrantSettings = new ImplicitGrantSettings();
+                }
+
+                if (toolOptions.EnableAccessToken.HasValue)
+                {
+                    updatedApp.Web.ImplicitGrantSettings.EnableAccessTokenIssuance = toolOptions.EnableAccessToken.Value;
+                }
+
+                if (toolOptions.EnableIdToken.HasValue)
+                {
+                    updatedApp.Web.ImplicitGrantSettings.EnableIdTokenIssuance = toolOptions.EnableIdToken.Value;
+                }
+
+                // TODO: update other fields. 
+                // See https://github.com/jmprieur/app-provisonning-tool/issues/10
+                try
+                {
+                    await graphServiceClient.Applications[existingApplication.Id]
```

Comment:
Theoretically, we make an unnecessary network call here if there were no changes to make.  Not to fix now, but it's Right to avoid that call if there were no factually no changes.

Reply:
Noted. Item to track #1560